soc/arm/st_stm32/stm32g0/Kconfig.defconfig.stm32g0b1xx

```diff
@@ -11,4 +11,8 @@ config SOC
 config NUM_IRQS
 	default 31
 
+config SHARED_INTERRUPTS
+	depends on CAN
+	default y if $(dt_nodelabel_enabled,fdcan1) && $(dt_nodelabel_enabled,fdcan2)
+
```

Comment:
```suggestion

if CAN_STM32_FDCAN

config SHARED_INTERRUPTS
	default y if $(dt_nodelabel_enabled,fdcan1) && $(dt_nodelabel_enabled,fdcan2)

endif # CAN_STM32_FDCAN
```

Reply:
Done 👍

Reply:
@henrikbrixandersen In General, recommendation used to be using `depends on FOO` when only one symbol is impacted and `if FOO` when more than one are impacted by the condition.

Also, in general I prefer to keep `BAR_STM32` symbols as much possible within the BAR driver. This is strictly equivalent and quicker to understand:
-`CAN` is well.. `CAN`
-`CAN_STM32_FDCAN` can suggest this is not exactly `CAN`, but some other conditions come into play ? So one would need to check

Reply:
From what I understood and what I've just tested right now with an example:

With the current code, If I compile any program that does not use CAN, I have the SHARED_INTERRUPT enabled (we don't want that) so the condition if indeed needed.

When CONFIG_CAN=y, the flag CONFIG_STM32_FDCAN is by default set as well only if you have at least one controller enabled in your dts.

In the end, both solution works but:
 - Using a dependency with CAN is cleaner but only rely on the fact CONFIG_STM32_FDCAN is enabled by default when CAN is enabled.
 - Using a dependency with CONFIG_STM32_FDCAN is more accurate as right now, the shared IRQ is only for the STM32_FDCAN driver. 

I leave it up to you guys

Reply:
> @henrikbrixandersen In General, recommendation used to be using `depends on FOO` when only one symbol is impacted and `if FOO` when more than one are impacted by the condition.

That is correct, but we are not defining a config option here, we are modifying one. We don't want to modify its dependencies to include `CONFIG_CAN` (users should be able to turn this on support for shared IRQs even if not using CAN).

> Also, in general I prefer to keep `BAR_STM32` symbols as much possible within the BAR driver. This is strictly equivalent and quicker to understand: -`CAN` is well.. `CAN` -`CAN_STM32_FDCAN` can suggest this is not exactly `CAN`, but some other conditions come into play ? So one would need to check

Well, you could have `CONFIG_CAN=y` but `CONFIG_CAN_STM32_FDCAN=n` if using an external CAN controller driver only, which again means that there is no need for changing the default of `CONFIG_SHARED_INTERRUPTS`.

Reply:
> * Using a dependency with CAN is cleaner but only rely on the fact CONFIG_STM32_FDCAN is enabled by default when CAN is enabled.

... and `CONFIG_STM32_FDCAN` can be disabled by the user.

> * Using a dependency with CONFIG_STM32_FDCAN is more accurate as right now, the shared IRQ is only for the STM32_FDCAN driver.

Exactly.